### PR TITLE
Add Gitlab MR support

### PR DESCRIPTION
**Pull request by ckaushik**
Add support for Merge Requests on Gitlab repositories

### CODE REVIEW

File: gitlab-mr-status/src/main/resources/gitlab/plugin.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<go-plugin id="gitlab.fb.status" version="1">
+    <about>
```
Comment (ketan) [inline gitlab-mr-status/src/main/resources/gitlab/plugin.xml]: The other plugins seem to use a template for this, can we be consistent?

https://github.com/gocd-contrib/gocd-build-status-notifier/blob/master/gerrit-cs-status/src/main/resources/plugin.xml

File: README.md [hunk 2]
```diff
@@ -102,9 +102,19 @@ Eg:
...
+
 ## FAQs
 
 [1]: images/list-plugin.png  "List Plugin"
 [2]: images/configure-plugin.png  "Configure Plugin"
-[3]: images/pipeline-schedule.png  "Pipeline Schedule"
-[4]: images/update-status.png  "On Successful Status Update"
+[3]: images/gitlab-plugin-settings.png  "Configure Gitlab Plugin"
+[4]: images/pipeline-schedule.png  "Pipeline Schedule"
+[5]: images/update-status.png  "On Successful Status Update"
```
Comment (ketan) [inline README.md]: `5` seems unused in the readme?

File: README.md [hunk 3]
```diff
@@ -102,9 +102,19 @@ Eg:
...
+```
+
 ## FAQs
 
 [1]: images/list-plugin.png  "List Plugin"
 [2]: images/configure-plugin.png  "Configure Plugin"
-[3]: images/pipeline-schedule.png  "Pipeline Schedule"
-[4]: images/update-status.png  "On Successful Status Update"
+[3]: images/gitlab-plugin-settings.png  "Configure Gitlab Plugin"
+[4]: images/pipeline-schedule.png  "Pipeline Schedule"
```
Comment (ketan) [inline README.md]: The images seem to be swapped out. See the image for point `When the stage status changes` and `The PR state is updated` [here](https://github.com/ckaushik/gocd-build-status-notifier/blob/master/README.md) and [here](https://github.com/gocd-contrib/gocd-build-status-notifier/blob/master/README.md)

File: gitlab-mr-status/src/main/resources/gitlab/plugin.xml (hunk 1 above)
Comment (ckaushik) [reply gitlab-mr-status/src/main/resources/gitlab/plugin.xml]: updated

File: README.md (hunk 3 above)
Comment (ckaushik) [reply README.md]: fixed this

File: README.md (hunk 2 above)
Comment (ckaushik) [reply README.md]: fixed the order and referenced the image in Gitlab config section

**Pull request merged**